### ISSUE TITLE
[FEATURE REQUEST] Filter Active text

### ISSUE THREAD

**Issue by GTEM19889** [notepad-plus-plus/notepad-plus-plus#5006]
On old mainframe systems I used to use a text editor (then known as Roscoe) that allowed me to filter all lines in my active document with a selection of text, and then edit it in place whilst the filter is active. I thought was the most useful tool ever but I am surprised to see that this is not in Notepad++ as standard (not as additional plugin)

Especially where let's say all the lines with text look exactly the same then it makes it easy to just make changes in place. At present I find myself copy and pasting everything into Excel where I can filter (but editing can be slightly more awkward) then making changes and copying back into Notepad which is a pain.

Might add to suggestion, to somehow illustrate that filter is applied either by altering colour or by adding very fine dotted line between lines that are filtered to show there are missing lines.

Thats all :)

**Comment by sasumner** [comment]
I understand what you are saying, as I have used an editor that did that (can't remember its name) in the past.  I found it an OK feature, but not a killer feature (at least I don't recall ever thinking that while I used it).

Notepad++ has a "hide lines" feature, but unfortunately I don't think it works very well.  And it isn't tied into search so that something like the filtering you mention could work.  But I think if "hide lines" was improved and actually tied into search, it could form the basis of the feature you are asking for.  My 2c.

Probably what most people do, rather than something drastic like copy-to-excel-and-back (ouch, it does indeed sound painful), is to make do with the Find All in Current Document feature and the output of the Find result panel.  While it isn't edit-in-place for sure, it basically provides pointers to those local places.  (I'm sure you know about this feature, it just isn't quite to the level of what you want).

So bottom line is I think you have a good idea here and I'd like to see something like this myself.

BTW, your "not as...plugin" comment made me think that you have a plugin in mind that already does this?

**Comment by GTEM19889** [comment]
Thanks for your response. 

Hide the lines means I have to select each line and hide it individually which makes going to excel the better solution.

Find all in current (Which I am aware of) can sometimes work, depending on the number of lines that I need to work with)  But I regularly find I need to edit either html or xml or even powershell where I want to quickly change a lot of lines containing the same word. Sometimes using the block edit (Alt and drag mouse) to overtype just one character in sections I have selected. So I revert to hopping back and forth between Excel and Notepad++ to achieve this. 

I thought it might not be too difficult to add a filter tab to the search box to allow one to filter content.

PS. Suggesting it should not be an add-on, is purely because I was expecting somebody to say this feature or almost similar is already available in add on. But quite often I need to edit something on a server and then I just download and install Notepad++ quickly, to then have to remind myself to everytime I do this go and look for a plugin and install that too, will simply be a pain.

I realise this might not be something everybody would want, although there is a part of me that thinks that people that have never used this will not know how useful it can be until they have used it a few times. I have often missed this feature.
